### PR TITLE
Prometheus: Metrics browser

### PR DESCRIPTION
**Pull request by davkal**
Similarly to the Logs browser for Loki (#30351), this PR adds a Metrics browser for Prometheus.

Kapture 2021-05-10 at 17 35 32

Notes for reviewer:

Ideally this still makes it into the beta 8. The metric (`__name__`) are a special label and are treated specially. As a result, I eschewed modifying the LogLabelBrowser, since this would have meant conditional code all over the place. I'd like to revisit this decision later, and either make it composable, or extract more common logic. 

What's missing now (compared to the Metrics cascader) is the Histogram and Recording rules groups, see Todos below.

Todos for later PRs (**moved to epic: #33976**):

- [x] add HELP and TYPE
- [ ] make metrics collapsible
- [ ] add histograms helper (search {le=~".+"} first?)
- [ ] add recording rules helper (or filter)
- [ ] investigate the new Labels APIs that recently started taking a selector (to replace the use of the Series API)

### CODE REVIEW

**Comment by ifrost** [comment]
Previously recording rules and histograms were grouped separately:

Screenshot 2021-05-11 at 14 55 55

Is there a way to access it now? There's a way to run "rate query" with a button at the bottom, but I can't find a way to get histogram query as before. Is it somewhere in the UI? Or maybe there should be an extra "run histogram query" button? (or should it be figured out automagically based on metric types)

File: public/app/plugins/datasource/prometheus/components/PromQueryField.test.tsx [hunk 1]
```diff
@@ -88,13 +90,12 @@ describe('PromQueryField', () => {
...
-    let cascader = screen.getByRole('button');
-    expect(cascader.textContent).toContain('Loading');
-    checkMetricsInCascader(await screen.findByRole('button'), changedMetrics);
+    // If we check the label browser right away it should be in loading state
+    let labelBrowser = screen.getByRole('button');
+    expect(labelBrowser.textContent).toContain('Loading');
   });
 
-  it('does not refreshes metrics when after rounding to minute time range does not change', async () => {
+  it.skip('does not refreshes metrics when after rounding to minute time range does not change', async () => {
```
Comment (ifrost) [inline public/app/plugins/datasource/prometheus/components/PromQueryField.test.tsx:98]: Why is it skipped?

File: public/app/plugins/datasource/prometheus/components/PromQueryField.tsx [hunk 2]
```diff
@@ -42,13 +40,14 @@ function getChooserText(metricsLookupDisabled: boolean, hasSyntax: boolean, metr
...
-  if (metrics && metrics.length === 0) {
+  if (!hasMetrics) {
     return '(No metrics found)';
   }
 
-  return 'Metrics';
+  return 'Metrics browser';
 }
 
+/* TODO
```
Comment (ifrost) [inline public/app/plugins/datasource/prometheus/components/PromQueryField.tsx:50]: Is it still needed? Can the commented code-block be removed?

File: public/app/plugins/datasource/prometheus/components/PromQueryField.tsx [hunk 3]
```diff
@@ -294,13 +284,14 @@ class PromQueryField extends React.PureComponent<PromQueryFieldProps, PromQueryF
...
     } = this.props;
-    const { histogramMetrics, metrics, metricsMetadata } = languageProvider;
+    const { metrics } = languageProvider;
 
     if (!metrics) {
       return;
     }
 
-    // Build metrics tree
+    /* TODO
```
Comment (ifrost) [inline public/app/plugins/datasource/prometheus/components/PromQueryField.tsx:293]: Can it be removed?

**Comment by ifrost** [comment]
> (...) As a result, I eschewed modifying the LogLabelBrowser, since this would have meant conditional code all over the place. I'd like to revisit this decision later, and either make it composable, or extract more common logic. (...)

Label.tsx and LokiLabel.tsx are almost identical. I think the only difference is the autoescape property? Could autoescape be set for Prometheus too? If yes we could make it a shared component. For the browser component - there's more tweaks in it but there's a lot of common code so it'd be great to make it a more generic component at some point. Maybe other data sources could use it too? I can imagine something like this for Graphite would be useful (and some users asked about it).

**Review by ifrost — APPROVED** [review]
It's good to go! Added an epic to address remaining issues #33976.

**Comment by oddlittlebird** [comment]
Is there documentation for this yet?

**Pull request merged**